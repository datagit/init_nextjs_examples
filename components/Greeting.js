import React from 'react';
import PropTypes from 'prop-types';

const Greeting = (props) => {
    return (
        <div>
            <h1>Hi {props.name}</h1>
        </div>
    );
}

// default value for props
Greeting.defaultProps = {
    name: 'default name!',
}

// set props type
Greeting.propTypes = {
    name: PropTypes.string.isRequired,
};
export default Greeting;