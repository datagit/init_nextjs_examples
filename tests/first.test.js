import React from 'react'
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Greeting from '../components/Greeting';

test('renders components Greeting', () => {
  const { getByText } = render(<Greeting />)
  const linkElement = getByText(
    /Hi default name!/
  )
  expect(linkElement).toBeInTheDocument();
})

it('renders components Greeting to create snapshot', () => {
  const tree = renderer.create(<Greeting />).toJSON()
  expect(tree).toMatchSnapshot()
})
