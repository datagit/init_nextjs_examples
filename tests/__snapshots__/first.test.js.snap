// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders components Greeting to create snapshot 1`] = `
<div>
  <h1>
    Hi 
    default name!
  </h1>
</div>
`;
